iorgen/parser_ocaml.py

```diff
@@ -69,28 +69,27 @@ def read_line(type_: Type, input_data: Input) -> str:
     # pylint: disable=too-many-return-statements
     assert type_.fits_in_one_line(input_data.structs)
     if type_.main == TypeEnum.INT:
-        return 'Scanf.scanf "%d\\n" (fun x -> x)'
+        return 'read_int ()'
     if type_.main == TypeEnum.CHAR:
-        return 'Scanf.scanf "%c\\n" (fun x -> x)'
+        return '(read_line ()).[0]'
     if type_.main == TypeEnum.STR:
-        return 'Scanf.scanf "%s@\\n" (fun x -> x)'
+        return 'read_line ()'
     if type_.main == TypeEnum.LIST:
         assert type_.encapsulated is not None
         if type_.encapsulated.main == TypeEnum.INT:
-            return ('Scanf.scanf "%s@\\n" (fun x -> {}List.map int_of_string '
-                    '(String.split_on_char \' \' x))'
-                    ).format('if String.equal "" x then [] else ' if type_.
-                             can_be_empty else "")
+            return ('read_line () |> fun x -> if x = "" then [] else String.split_on_char \' \' x'
```

Comment:
You do not like the idea of removing `if x = "" then [] else ` if it is not necessary to specify it?

Reply:
For input reading I prefer to be more careful than necessary. And it simplifies the code of iorgen a little bit.